### PR TITLE
feat: add preflights check for HyperV

### LINKED ISSUE #6207: Add checks for Hyper-V in Podman extension
### Is your enhancement related to a problem? Please describe

as part of the epic https://github.com/containers/podman-desktop/issues/5113 we should have preflight checks to see if Hyper-V is installed or not

### Describe the solution you'd like

check if hyper-v is installed or not

we should check for WSL and Hyper-V and then creating the machine should be a Hyper-V or WSL depending on the supported platform

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jeffmaury**
Fixes #6207

### What does this PR do?

Add a new preflight check for HyperV

### Screenshot / video of UI

N/A

### What issues does this PR fix or reference?

#6207 

### How to test this PR?

Run Podman extension unit test or remove WSL and HyperV and install Podman

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

**Review by benoitf — COMMENTED** [review]
maybe introduce SequenceCheck and OrCheck in a separate PR, then use it
(and add tests for these 2 new objects)

also I'm not sure why isUserAdmin is moved to a pure function (looks like it can be in a WindowsCheck class extending BaseCheck

**Review by benoitf — APPROVED** [review]
code LGTM but as testing on macOS the code is not used, so probably need a review from someone on Windows

File: extensions/podman/src/podman-install.ts [hunk 1]
```diff
@@ -807,3 +816,61 @@ class WSLVersionCheck extends BaseCheck {
...
+      ]);
+      return result.stdout === 'True';
+    } catch (err: unknown) {
+      return false;
+    }
+  }
+
+  private async isHyperVRunning(): Promise<boolean> {
+    try {
+      const result = await extensionApi.process.exec('powershell.exe', ['@(Get-Service vmms).Status']);
```
Comment (feloy) [inline extensions/podman/src/podman-install.ts]: On my Windows machine, I get `Running` as output of this command, not `True`

**Review by feloy — COMMENTED** [review]
on my machine, I uninstalled HyperV and WSL. 

I started Podman Desktop, the check indicates that Hyoer-V is not running.

I follow the instructions to install it, reboot, and check that it is running (`(Get-Service vmms).Status` is Running)

I restart Podman Desktop, but it still says that HyperV is not running

File: extensions/podman/packages/extension/src/podman-install.ts [hunk 2]
```diff
@@ -807,3 +816,61 @@ class WSLVersionCheck extends BaseCheck {
...
+        },
+      });
+    }
+    return this.createSuccessfulResult();
+  }
+
+  private async isHyperVinstalled(): Promise<boolean> {
+    try {
+      const result = await extensionApi.process.exec('powershell.exe', [
+        '@(Get-Wmiobject Win32_ComputerSystem).HypervisorPresent',
```
Comment (feloy) [inline extensions/podman/packages/extension/src/podman-install.ts]: I have uninstalled hyperv (from the Turn Windows features on or off), rebooted, but this command still returns True

Comment (jeffmaury) [reply extensions/podman/packages/extension/src/podman-install.ts]: Good catch, updated

**Comment by feloy** [comment]
The preflights LGTM: starting with neither wsl not hyperV installed,  the preflight makes me install Hyper-V, then pass.

But then, when I try to create a machine, I get an error that `wsl` is not found, but I would expect that it is created with Hyper-V (as only Hyper-V is installed on my system)

**Comment by jeffmaury** [comment]
> The preflights LGTM: starting with neither wsl not hyperV installed, the preflight makes me install Hyper-V, then pass.
> 
> But then, when I try to create a machine, I get an error that `wsl` is not found, but I would expect that it is created with Hyper-V (as only Hyper-V is installed on my system)

This PR is only about adding the checks
Machine creation and query is another PR

**Review by feloy — APPROVED** [review]

**Pull request merged**